### PR TITLE
062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_perm…

### PR DESCRIPTION
**Pull request by istoimenova**
Script is part of TS: #932

### CODE REVIEW

**Comment by istoimenova** [comment]
@OVikhrov, @dboltovskyi, please review.

**Comment by dboltovskyi** [comment]
@istoimenova, Since this script is almost the same as 060 review is on Hold until review of 060 is finished

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+testCasesForPolicyTable:precondition_updatePolicy_AllowFunctionInHmiLeves({"NONE", "LIMITED", "BACKGROUND", "FULL"}, "SetAudioStreamingIndicator")
+
+--[[ Local Functions ]]
+
+--[[@create_ptu_SetAudioStreamingIndicator_group: creates PTU files
+! SetAudioStreamingIndicator_group1.json: assign application with null permissions.
+! @parameters: NO
+]]
+local function create_ptu_SetAudioStreamingIndicator_group()
+  os.execute(" cp " .. config.pathToSDL .. "sdl_preloaded_pt.json" .. " " .. "files/SetAudioStreamingIndicator_group1.json" )
```
Comment (OVikhrov) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @istoimenova Please use commonPreconditions:GetPathToSDL() for getting SDL path

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  local pathToFile = config.pathToSDL .. 'sdl_preloaded_pt.json'
+  local file = io.open(pathToFile, "r")
+  local json_data = file:read("*all")
+  file:close()
+
+  local data = json.decode(json_data)
+  if(data.policy_table.functional_groupings["DataConsent-2"]) then
+    data.policy_table.functional_groupings["DataConsent-2"].rpcs = { json.null }
```
Comment (OVikhrov) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @istoimenova Please use

> data.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null

to get correct preloaded in view:

>         "DataConsent-2": {
>             "user_consent_prompt": "DataConsent",
>             "rpcs": null
>         },

In other case you will get:

>   	"DataConsent-2": {
>   		"rpcs": [null],
>   		"user_consent_prompt": "DataConsent"
>              }

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  end
+  data.policy_table.module_config.preloaded_pt = nil
+  data.policy_table.module_config.preloaded_date = nil
+  --TODO(istoimenova): Update when problem with saving in file: "0000001": null using json.null.
+  -- data.policy_table.app_policies[config.application1.registerAppInterfaceParams.appID] = json.null
+  data.policy_table.app_policies[config.application1.registerAppInterfaceParams.appID] = null
```
Comment (OVikhrov) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @istoimenova Plese use json.null

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_ActivateApp()
+  commonSteps:ActivateAppInSpecificLevel(self, self.applications[config.application1.registerAppInterfaceParams.appName])
+  EXPECT_NOTIFICATION("OnHMIStatus", {systemContext = "MAIN", hmiLevel = "FULL"})
+end
+
+function Test:Precondition_PolicyTableUpdate_Proprietary()
+  local SystemFilesPath = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath")
```
Comment (OVikhrov) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @istoimenova Please add pull request to update in commonFunctions function read_parameter_from_smart_device_link_ini to using commonPreconditions:GetPathToSDL().
In another case test woll be failed on  CI

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua (hunk 1 above)
Comment (istoimenova) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @OVikhrov, done in 15a5682391d8a2bced732b304a4fefbde1bbde7c

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua (hunk 2 above)
Comment (istoimenova) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @OVikhrov, done in 15a5682

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua (hunk 3 above)
Comment (istoimenova) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @OVikhrov, done in 15a5682

File: test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua (hunk 4 above)
Comment (istoimenova) [inline test_scripts/API/SetAudioStreamingIndicator/062_ATF_N_TC_SetAudioStreamingIndicator_DISALLOWED_navi_app_null_permission.lua]: @OVikhrov, PR is: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1063.
I have updated this in all places in file. You are assigned as reviewer.

**Pull request merged**